text/0141-linking-traces.md

```diff
@@ -0,0 +1,447 @@
+- Start Date: 2024-11-08
+- RFC Type: feature 
+- RFC PR: [#141](https://github.com/getsentry/rfcs/pull/141)
+- RFC Status: draft
+
+# Summary
+
+This RFC introduces the ability to link and connect traces in Sentry. The required changes involve SDKs, Ingest, Search and Storage, Backend as well as product teams.
+The goal of this initiative is to be able to link multiple traces together so that the Sentry product can show what happened before a specific trace while preserving the integrity of individual traces.
+This RFC proposes to use [Span Links](#preferred-span-links) as a vehicle for specifying relationships between spans across multiple traces. It also discusses other options considered, as well as concrete use cases for this feature.
+
+# Motivation
+
+At the time of writing this RFC, Sentry (the product as well as SDKs) cannot connect multiple traces to deliver a "bigger picture" of what happened before (or after) a specific trace. 
+A trace is therefore the highest level of connecting signals (errors, traces, profiles, etc.).
+The notable exception is Session Replay, where we are indeed able to provide this bigger picture.
+Replay is an opt-in, additional product.
+
+This RFC proposes a method of achieving linkage between (and within) traces in the entire Sentry product, to satisfy a number of currently sub-optimally solved use cases.
+
+## Frontend Traces and User Journeys
+
+The most important application of linking traces are frontend applications. We would like to better display a user journey (session) to make debugging of issues easier as developers get more context on what happened before a specific issue. 
```

Comment:
Just writing down a thought here: I think we have multiple definitions of "session" in our product that we should be aware of and be careful with in the communication. The ones I have on top of my mind: replay sessions, release health sessions and user journey sessions in tracing.

Reply:
While reviewing, I found another related comment on this further down in this file: https://github.com/getsentry/rfcs/pull/141/files#r1856179314

Reply:
I removed the "session" reference in this section. As you pointed out, I described the two current sessions in the SDK further below, where we also point out that trace links are not another session but upon leadership decision an alternative to a 3rd session.